torch/csrc/jit/codegen/cuda/lower_utils.cpp

```diff
@@ -277,6 +337,195 @@ std::pair<kir::ForLoop*, int64_t> getAllocPoint(
 }
 
 } // namespace loop_utils
+
+namespace kir_utils {
+
+namespace {
+
+class ReplaceExprInput : public kir::MutableIrVisitor {
+ public:
+  static kir::Expr* replace(
+      kir::Expr* expr,
+      const std::unordered_map<kir::Val*, kir::Val*>& replacement_map) {
+    ReplaceExprInput replacer(expr, replacement_map);
+    expr->accept(&replacer);
+    TORCH_INTERNAL_ASSERT(expr != nullptr);
```

Comment:
nit: If this needs to be checked, it should be placed before the call to `accept`.